CMakeLists.txt

```diff
@@ -27,6 +27,78 @@ ELSE()
     SET(VER_BUILD 0)
 ENDIF()
 
+# Configuration of compilers
+## Compiler flags to add
+### Flags for GCC and Clang
+set(GNU_FLAGS_WARNING_YES "-Wall -Wextra -Wformat=2 -Wpedantic -Wshadow -Wmissing-declarations -Wfloat-equal")
+set(GNU_FLAGS_WARNING_NO  "-Wno-unused-command-line-argument -Wno-documentation -Wno-format-nonliteral")
+set(GNU_FLAGS_WARNING     "${GNU_FLAGS_WARNING_YES} ${GNU_FLAGS_WARNING_NO}")
+set(GNU_FLAGS_ERROR       "-pedantic-errors -Werror") # warning is error
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU" OR
+    CMAKE_C_COMPILER_ID STREQUAL "GCC")
+  set(GNU_FLAGS_SUGGEST   "-Wsuggest-attribute=pure")
+endif()
+set(GNU_FLAGS_DEFAULT     "${GNU_FLAGS_WARNING} ${GNU_FLAGS_ERROR} ${GNU_FLAGS_SUGGEST}")
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU"   AND
+    CMAKE_C_COMPILER_VERSION VERSION_GREATER 4.9)
+  # A color output is easier to read.
+  set(GNU_FLAGS_DEFAULT   "${GNU_FLAGS_DEFAULT} -fdiagnostics-color=auto")
```

Comment:
Please remove the diagnostics color as it is done per default in recent gcc, right?